### ISSUE TITLE
Lexing problem

### ISSUE THREAD

**Issue by zetachang** [opal/opal#1074]
Stuff like this could not be correctly tokenized.

```ruby
10
b = `2.2204460492503130808472633361816E-16`
```
**[Try It](http://opalrb.org/try/?code:10%0Ab%20%3D%20%602.2204460492503130808472633361816E-16%60%0A)**

After investigation, problem is https://github.com/opal/opal/blob/master/lib/opal/parser/lexer.rb#L215, the "^" regexp provided by JS implemented "strscan" will only be applied to the first alternative, thus the later one will be used to match and we will got a wrong token.

**Comment by zetachang** [comment]
Forget to add an example demonstrating the failing case of StringScanner.

```ruby
require "strscan"
scanner = StringScanner.new("10\nb = `2E-16`")
puts scanner.scan(/[\d_]+\.[\d_]+\b|[\d_]+(\.[\d_]+)?[eE][-+]?[\d_]+\b/)
```

MRI output nil, while Opal output `2E-16`

**Comment by alexdowad** [comment]
@zetachang, thanks for providing the failing example. I used it for a test case.

Please have a look at the `strscanner_scan_fix` branch which I just pushed. Look good?

**Comment by alexdowad** [comment]
Everything is green, so I'm merging. Fixed!

**Comment by elia** [comment]
(fixed by e201d8050eb07e8bbedb8171688aca5178f2dd26)
